### PR TITLE
Add an option to only throw exception on 5XX error codes

### PR DESCRIPTION
**Pull request by joelwurtz**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #77 
| Documentation   | https://github.com/php-http/documentation/pull/235/
| License         | MIT

cf #77

### CODE REVIEW

**Comment by joelwurtz** [comment]
> I like this feature. Do you think it is a good idea (or overkill) to let the user provide a regex to decide when to throw an exception?

I think it's overkill, someone can still do it's own plugin for that, however handling 4XX responses is common (like you want the bad request for handling validations, or 404 so you know that this resource does not exist and can handle it properly, but 5XX responses are never wanted because client don't know how to respond to them)

**Comment by sagikazarmark** [comment]
Like the idea. Is this PR ready?

**Comment by joelwurtz** [comment]
For me yes

**Comment by sagikazarmark** [comment]
Great. Can you write some documentation as well?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,15 @@
 # Change Log
 
-## Upcoming
+## 1.8 (unreleased)
```
Comment (GrahamCampbell) [inline CHANGELOG.md]: 1.8.0?

Comment (dbu) [reply CHANGELOG.md]: i guess we can still add the .0 when we do the release

**Pull request merged**